stdlib/xml/sax/__init__.pyi

```diff
@@ -34,7 +34,9 @@ else:
     def make_parser(parser_list: list[str] = ...) -> XMLReader: ...
 
 def parse(
-    source: str | _SupportsReadClose[bytes] | _SupportsReadClose[str], handler: ContentHandler, errorHandler: ErrorHandler = ...
+    source: StrPath | _SupportsReadClose[bytes] | _SupportsReadClose[str],
```

Comment:
Shouldn't `source` also take `xmlreader.InputSource` ? `source` is coerced into an `InputSource` and the docstring also seems to imply one can be passed directly.
https://github.com/python/cpython/blob/3.11/Lib/xml/sax/saxutils.py#L338
https://github.com/python/cpython/blob/3.11/Lib/xml/sax/expatreader.py#L105
https://github.com/python/cpython/blob/3.11/Lib/xml/sax/xmlreader.py#L117

@srittau Dunno if you wanna consider this outside the scope of this PR, because the type for parameter `source` is missing in a few locations. But I wanted to at least raise this.